R/deleteThis.R

```diff
@@ -0,0 +1,1789 @@
+# You can pass it objects (and a progressbar) to be exported to the cluster.
+# To be used in combination with the foreach package.
+.makeParallelSetup <- function(pb=NULL, objs=NULL, env=NULL) {
+
+  nCores <- parallel::detectCores(TRUE) - 1
+  if (is.na(nCores) || nCores == 0)
+    nCores <- 1
+
+  cl <- snow::makeSOCKcluster(nCores)
+  doSNOW::registerDoSNOW(cl)
+  if (! is.null(objs) && ! is.null(env))
+    snow::clusterExport(cl, objs, envir=env)
+
+  dopar <- foreach::`%dopar%`
+
+  progress <- NULL
+  if (! is.null(pb))
+    progress <- function() .updateParallelProgressbar(pb, cl)
+
+  stopCluster <- substitute(try(snow::stopCluster(cl), silent=TRUE))
+
+  return(list(cl=cl, progress=list(progress=progress), dopar=dopar, stopCluster=stopCluster))
+}
+
+.modifyStateFigures <- function(x, identifier, replacement, completeObject = TRUE) {
+
+  # recursive function that traverses the entire state object to replace old figures with new figures
+  # it searches a list where lst[["data"]] == identifier and then does lst[["obj"]] <- replacement
+  # TODO: also add specific modifications for writeImage?
+
+  # completeObject: boolean. If TRUE, then the complete sublist identified by the identifier will
+  #                 be replaced by replacement. If FALSE, replacement should be a list where the
+  #                 elements will be looked up
+
+  # check if list
+  if (inherits(x, "list")) { # not is.list to avoid false positive (i.e. ggplot objects are lists)
+
+    # check if plotting list we're looking for
+    # if (!is.null(x[["editable"]]) && x[["data"]] == identifier) {
+    if (!is.null(x[["data"]]) && x[["data"]] == identifier) {
+
+      if (!completeObject) {
+        x[names(replacement)] <- replacement
+        return(x)
+      } else {
+        return(replacement)
+      }
+
+    } else {
+      # check if criteria are met in any of the sublists
+      return(lapply(x, .modifyStateFigures, identifier = identifier, replacement = replacement, completeObject = completeObject))
+    }
+  } else {
+    # return ordinary object
+    return(x)
+  }
+}
+
+.getFigureFromState <- function(x, identifier) {
+
+  # recursive function that traverses the entire state object to find a plot by the png filename (identifier).
+  # returns all matches found
+  if (is.list(x)) {
+    if (identical(x[["data"]], identifier)) {
+      return(x)
+    } else {
+      return(unlist(lapply(unname(x), .getFigureFromState, identifier), recursive = FALSE))
+    }
+  }
+}
+
+.quietDuringUnitTest <- function(expr) {
+  # check from testthat::skip_on_travis
+  if (identical(Sys.getenv("TRAVIS"), "true")) {
+    return(suppressWarnings(suppressMessages(expr)))
+  } else {
+    return(expr)
+  }
+}
+
+# Operator to perform an expression by group / conditionally on a factor / given a condition. (author: EJvK):
+# Usage: (abs(mtcars$mpg - mean(mtcars$mpg)) > var(mtcars$mpg)) %|% mtcars$cyl
+`%|%` <- function(expr, group)
```

Comment:
This one appears to be used by the filter/compute column interface

Reply:
I've added it to `friendlyConstructorFunctions.R` which is a somewhat better home for it than `common.R`.

Reply:
Yup sounds sensible